arangod/Network/ConnectionPool.cpp

```diff
@@ -199,8 +217,12 @@ std::shared_ptr<fuerte::Connection> ConnectionPool::createConnection(fuerte::Con
 
 ConnectionPtr ConnectionPool::selectConnection(std::string const& endpoint,
                                                ConnectionPool::Bucket& bucket) {
+
+  using namespace std::chrono;
   std::lock_guard<std::mutex> guard(bucket.mutex);
 
+  auto start = high_resolution_clock::now();
```

Comment:
from https://en.cppreference.com/w/cpp/chrono/high_resolution_clock:

> The high_resolution_clock is not implemented consistently across different standard library implementations, and its use should be avoided. It is often just an alias for std::chrono::steady_clock or std::chrono::system_clock, but which one it is depends on the library or configuration. When it is a system_clock, it is not monotonic (e.g., the time can go backwards). For example, for gcc's libstdc++ it is system_clock, for MSVC it is steady_clock, and for clang's libc++ it depends on configuration.

Reply:
Let's rather use `steady_clock` here, so the clock cannot go backwards and negative durations are avoided.

Reply:
Will do.